Formula/aws-session-manager-plugin.rb

```diff
@@ -0,0 +1,18 @@
+class AwsSessionManagerPlugin < Formula
+  desc "Official Amazon AWS session manager plugin"
+  homepage "https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html"
+  url "https://s3.amazonaws.com/session-manager-downloads/plugin/1.1.35.0/mac/sessionmanager-bundle.zip"
+  version "1.1.35.0"
+  sha256 "9aa7c6372a39285ece98ad0bc18b23cc0924c6f1d9fb67e1ef360eb65e0653a8"
+
+  depends_on "awscli"
+
+  def install
+    bin.install "bin/session-manager-plugin"
+    prefix.install %w[LICENSE VERSION]
+  end
+
+  test do
+    system bin/"session-manager-plugin"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This plugin is to enable the `awscli` to open a shell session with an ec2 instance, running on AWS, that is running an agent.

It itself doesn't have any functionality.

The way to test the plugin works is to just run the plugin, as done in the test, as it will tell you if it is configured properly.